pkg/apis/serving/metadata_validation.go

```diff
@@ -30,7 +31,18 @@ import (
 // resources is correct.
 func ValidateObjectMetadata(meta metav1.Object) *apis.FieldError {
 	return apis.ValidateObjectMetadata(meta).
-		Also(autoscaling.ValidateAnnotations(meta.GetAnnotations()).ViaField("annotations"))
+		Also(autoscaling.ValidateAnnotations(meta.GetAnnotations()).
+			Also(validateKnativeAnnotations(meta.GetAnnotations())).
+			ViaField("annotations"))
+}
+
+func validateKnativeAnnotations(annotations map[string]string) (errs *apis.FieldError) {
+	for key := range annotations {
+		if strings.HasPrefix(key, GroupNamePrefix) && key != UpdaterAnnotation && key != CreatorAnnotation && key != RevisionLastPinnedAnnotationKey {
+			errs = errs.Also(apis.ErrInvalidKeyName(key, ""))
```

Comment:
```suggestion
			errs = errs.Also(apis.ErrInvalidKeyName(key, apis.CurrentField))
```
I think that's how it's spelled.